app/components/Header.tsx

```diff
@@ -84,7 +85,6 @@ export default function Header() {
               icon={hamburger ? <FaXmark /> : <FaBars />}
             />
           </Box>
-          <Spacer></Spacer>
```

Comment:
Per ChakraUI docs: "Since Spacer renders a div, any gap value provided to the parent is applied to both sides of this component, and therefore make the gap appear doubled when the spacer is completely collapsed." It also seems that it's supposed to be used with Flex component, not Stack; so maybe that also adds some strange behavior. This though causes the layout shift in the header that can be solved with adding padding/changing how you size image/playing with gap and grid sizing/using another component instead of the Stack